google-cloud-bigquery/test/google/cloud/bigquery/dataset_load_job_updater_storage_test.rb

```diff
@@ -0,0 +1,350 @@
+    dataset.service.mocked_service = mock
+
+    job = dataset.load_job table_id, load_file do |j|
+      _(j.location).must_equal "US"
+      j.location = region
+    end
+    mock.verify
+
+    _(job).must_be_kind_of Google::Cloud::Bigquery::LoadJob
+    _(job.location).must_equal region
+  end
+
+  it "can load a storage file and unset location" do
+    mock = Minitest::Mock.new
+    insert_job_gapi = load_job_url_gapi table_gapi.table_reference, load_url
+    return_job_gapi = load_job_url_gapi table_gapi.table_reference, load_url
+    insert_job_gapi.job_reference.remove_instance_variable :@location
+    return_job_gapi.job_reference.location = "US"
+    mock.expect :insert_job, return_job_gapi,
+                [project, insert_job_gapi]
+    dataset.service.mocked_service = mock
+
+    job = dataset.load_job table_id, load_file do |j|
+      _(j.location).must_equal "US"
+      j.location = nil
+    end
+    mock.verify
+
+    _(job).must_be_kind_of Google::Cloud::Bigquery::LoadJob
+    _(job.location).must_equal "US"
+  end
+
+  it "can specify a storage URI with hive partitioning options" do
+    gcs_uri = "gs://cloud-samples-data/bigquery/hive-partitioning-samples/autolayout/*"
+    source_uri_prefix = "gs://cloud-samples-data/bigquery/hive-partitioning-samples/autolayout/"
+
+    mock = Minitest::Mock.new
+    hive_partitioning_options = Google::Apis::BigqueryV2::HivePartitioningOptions.new(
+      mode: "AUTO",
+      source_uri_prefix: source_uri_prefix
+    )
+    job_gapi = load_job_url_gapi table_gapi.table_reference, gcs_uri, hive_partitioning_options: hive_partitioning_options
+    job_gapi.configuration.load.source_format = "PARQUET"
+    mock.expect :insert_job,
+                load_job_resp_gapi(table, gcs_uri, source_format: "PARQUET", hive_partitioning_options: hive_partitioning_options),
+                [project, job_gapi]
+    dataset.service.mocked_service = mock
+
+    job = dataset.load_job table_id, gcs_uri do |j|
+      j.format = "parquet"
+      j.hive_partitioning_mode = :auto
+      j.hive_partitioning_source_uri_prefix = source_uri_prefix
+    end
+    mock.verify
+
+    _(job).must_be_kind_of Google::Cloud::Bigquery::LoadJob
+    _(job.parquet?).must_equal true
+    _(job.hive_partitioning?).must_equal true
+    _(job.hive_partitioning_mode).must_equal "AUTO"
+    _(job.hive_partitioning_require_partition_filter?).must_equal false
+    _(job.hive_partitioning_source_uri_prefix).must_equal source_uri_prefix
+  end
+
+  # Borrowed from MockStorage, load to a common module?
+
+  def random_bucket_hash name=random_bucket_name
+    {"kind"=>"storage#bucket",
+     "id"=>name,
+     "selfLink"=>"https://www.googleapis.com/storage/v1/b/#{name}",
+     "projectNumber"=>"1234567890",
+     "name"=>name,
+     "timeCreated"=>::Time.now,
+     "metageneration"=>"1",
+     "owner"=>{"entity"=>"project-owners-1234567890"},
+     "location"=>"US",
+     "storageClass"=>"STANDARD",
+     "etag"=>"CAE=" }
```

Comment:
I guess we're not running rubocop on these test files? I think this hash literal would set off several warnings (space surrounding the brackets, space surrounding the arrows, alignment of values).

Reply:
The coding style of the implementation was never applied to tests in any of the veneer packages, unfortunately. :(

Reply:
I was afraid of that. Oh well. Not a huge priority for now.